src/contracts/storage/pool.rs

```diff
@@ -58,15 +56,31 @@ impl Default for Pool {
 }
 
 impl Pool {
-    pub fn create(init_tick: i32, current_timestamp: u64, fee_receiver: AccountId) -> Self {
-        Self {
-            sqrt_price: unwrap!(calculate_sqrt_price(init_tick)),
+    pub fn create(
+        init_sqrt_price: SqrtPrice,
+        mut init_tick: i32,
+        current_timestamp: u64,
+        tick_spacing: u16,
+        fee_receiver: AccountId,
+    ) -> Result<Self, InvariantError> {
+        let upper_bound = unwrap!(SqrtPrice::from_tick(init_tick));
+        let lower_bound = unwrap!(SqrtPrice::from_tick(init_tick - tick_spacing as i32));
```

Comment:
It will be better to calculate:
(1) `sqrt_price(init_tick)`
(2) `sqrt_price(init_tick + tick_spacing)`
then
sqrt_price must be in range <sqrt_price(init_tick), sqrt_price(init_tick + tick_spacing))